docs/rules/no-js-extension.md

```diff
@@ -36,6 +43,55 @@ define(function (require) {
     var foo = require('foo');
     /* ... */
 });
+
+require(['pluginname!foo'], function (foo) {
```

Comment:
By default this is _not_ a warning: require(['pluginname!foo'], function (foo) {

> Would it make more sense to have a separate rule for verifying that plugin-prefixed paths actually have an extension?

That's what the options array does. Paths which start with one of listed plugin-prefixes _are_ checked for `.js`

Or maybe I don't get your point.